src/dualscreen.tsx

```diff
@@ -95,29 +96,35 @@ export class DualScreen extends KalturaPlayer.core.BasePlugin {
     this._removeActivesArr.push(
       this._player.ui.addComponent({
         label: 'kaltura-dual-screen-pip',
-        presets: ['Playback', 'Live', 'Error', 'Ads', 'Idle'],
+        presets: ['Playback', 'Live', 'Ads'],
         container: ReservedPresetAreas.InteractiveArea,
         get: () => (
           <ResponsiveManager
             onMinSize={() => {
-              this._switchToPIPMinimized(false);
+              this._switchToPIPMinimized(false, Animations.None);
             }}
             onDefaultSize={this._setMode}>
-            <Pip
-              childSizePercentage={this.config.childSizePercentage}
-              childPlayer={this._secondaryKalturaPlayer}
-              position={this.config.position}
-              hide={this._switchToPIPMinimized}
-              onSideBySideSwitch={this._switchToSideBySide}
-              onInversePIP={this._switchToPIPInverse}
-            />
+            <DragAndSnapManager
+              eventManager={this.eventManager}
+              logger={this.logger}
+              getPosition={this.getPipPosition}
+              onPositionChanged={this._setPipPosition}>
+              <PipChild
+                animation={Animations.Fade}
+                playerSizePercentage={this.config.childSizePercentage}
+                player={this._secondaryKalturaPlayer}
+                hide={() => this._switchToPIPMinimized(true, Animations.None)}
+                onSideBySideSwitch={this._switchToSideBySide}
+                onInversePIP={() => this._switchToPIPInverse(true, Animations.Fade)}
+              />
+            </DragAndSnapManager>
           </ResponsiveManager>
         )
       })
     );
   };
 
-  private _switchToPIPInverse = (manualChange: boolean = true) => {
+  private _switchToPIPInverse = (manualChange: boolean, animation: Animations) => {
```

Comment:
When manualChange is false you send animations.none to the parent pip. Child pip never uses the animation parameter - So, first the name is confusing. Maybe call it parentAnimation. Second, I would give a default "none" and don't pass it at all when manualChange is false. And then actually you can remove completly the manualChange param as it is used only for the pipParent animation decision